### PR TITLE
Add swap disk as vdisk support

### PR DESCRIPTION
**Pull request by jichenjc**

### CODE REVIEW

File: smtLayer/makeVM.py [hunk 1]
```diff
@@ -73,7 +73,9 @@
         '--iplLoadparam': ['iplLoadparam', 1, 2],
         '--dedicate': ['dedicate', 1, 2],
         '--loadportname': ['loadportname', 1, 2],
-        '--loadlun': ['loadlun', 1, 2]},
+        '--loadlun': ['loadlun', 1, 2],
+        '--vdisk': ['vdisk', 1, 2],
+        '--vdisksize': ['vdisksize', 1, 2]},
```
Comment (bjhuangr) [inline smtLayer/makeVM.py:78]: `vdisksize` added but not used?

File: zvmsdk/api.py [hunk 2]
```diff
@@ -791,20 +800,23 @@ def guest_create(self, userid, vcpus, memory, disk_list=None,
...
-                # 'disk_pool' format check
-                if ':' not in disk_pool or (disk_pool.split(':')[0].upper()
-                    not in ['ECKD', 'FBA']):
-                    errmsg = ("Invalid disk_pool input, it should be in format"
-                              " ECKD:eckdpoolname or FBA:fbapoolname")
-                    LOG.error(errmsg)
-                    raise exception.SDKInvalidInputFormat(msg=errmsg)
+                if not swap_only:
+                    disk_pool = disk.get('disk_pool') or CONF.zvm.disk_pool
+                    if disk_pool is None and not swap_only:
```
Comment (bjhuangr) [inline zvmsdk/api.py:807]: already in `not swap_only` if condition

File: zvmsdk/smtclient.py [hunk 3]
```diff
@@ -614,6 +614,40 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
             if 'lun' in loaddev:
                 rd += ' --loadlun %s' % loaddev['lun']
 
+        # now, we need consider swap only case, customer using boot
+        # from volume but no disk pool provided, we allow to create
+        # swap disk from vdisk by default, when we come to this logic
+        # we are very sure that if no disk pool, there is only one
+        # disk in disk_list and that's swap
+        new_disk_list = []
+        vdisk = False
```
Comment (bjhuangr) [inline zvmsdk/smtclient.py:623]: `new_disk_list` and `vdisk` are not required. set `disk_list` to `[]` after `--vdisk` added into the `rd`.

**Comment by bjhuangr** [comment]
@dongyanyang It would be much helpful if you can join to review this PR.

File: smtLayer/makeVM.py (hunk 1 above)
Comment (jichenjc) [reply smtLayer/makeVM.py:78]: removed

File: zvmsdk/api.py (hunk 2 above)
Comment (jichenjc) [reply zvmsdk/api.py:807]: removed

File: zvmsdk/smtclient.py (hunk 3 above)
Comment (jichenjc) [reply zvmsdk/smtclient.py:623]: will adjust logic here, functional test find some issues

**Comment by bjhuangr** [comment]
As we discussed, can we add a config option to enable vdisk swap?

File: zvmsdk/smtclient.py [hunk 4]
```diff
@@ -636,13 +666,18 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
         with zvmutils.log_and_reraise_sdkbase_error(action):
             self._GuestDbOperator.add_guest(userid)
 
-        # Continue to add disk
-        if disk_list:
+        # Continue to add disk, if vdisk is None, it means
+        # it's not vdisk routine and we need add disks
+        if vdisk is None:
```
Comment (bjhuangr) [inline zvmsdk/smtclient.py:671]: handle the case that `disk_list == []`

File: zvmsdk/smtclient.py [hunk 5]
```diff
@@ -614,6 +614,36 @@ def create_vm(self, userid, cpu, memory, disk_list, profile,
...
+
+        if len(disk_list) == 1:
+            disk = disk_list[0]
+            if 'format' in disk and disk['format'].lower() == 'swap':
+                disk_pool = disk.get('disk_pool') or CONF.zvm.disk_pool
+                if disk_pool is None:
+                    # if it's vdisk, then create user direct directly
+                    vd = disk.get('vdev') or self.generate_disk_vdev(offset=0)
+                    disk['vdev'] = vd
+                    sizeUpper = disk.get('size').strip().upper()
```
Comment (dongyanyang) [inline zvmsdk/smtclient.py:632]: suggest to move these size handling into smtlayer, refer to add_mdisk which called the smtlayer: changeVM.

**Review by dongyanyang — COMMENTED** [review]
LGTM

**Review by bjhuangr — APPROVED** [review]

**Pull request merged**